Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift

```diff
@@ -0,0 +1,216 @@
+  case pluralEntry(PluralEntry)
+  case variableWidthEntry(VariableWidthEntry)
+}
+
+extension StringsDict {
+  /// Plural entries are used to define language plural rules.
+  ///
+  /// Reference: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/StringsdictFileFormat/StringsdictFileFormat.html
+  // swiftlint:disable:previous line_length
+  struct PluralEntry: Codable {
+    /// A format string that contains variables.
+    /// Each variable is preceded by the %#@ characters and followed by the @ character.
+    let formatKey: String
+    /// An array of variables specifying the rules to use for each variable in the `formatKey`.
+    /// Every variable that is part of the `formatKey` must be contained in this dictionary.
+    let variables: [Variable]
+  }
+
+  /// Variable width entries are used to define width variations for a localized string.
+  ///
+  /// Reference: https://developer.apple.com/documentation/foundation/nsstring/1413104-variantfittingpresentationwidth
+  struct VariableWidthEntry: Codable {
+    let rules: [String: String]
+  }
+}
+
+extension StringsDict.PluralEntry {
+  /// `Variable`s are a key-value pair specifying the rule to use for each variable (key).
+  struct Variable: Codable {
+    let key: String
+    let rule: VariableRule
+  }
+
+  /// A VariableRule contains the actual localized plural strings for every possible plural case that
+  /// is available in a specific locale. Since some locales only support a subset of plural forms,
+  /// most of them are optional.
+  struct VariableRule: Codable {
+    /// The only possible value is `NSStringPluralRuleType`.
+    let specTypeKey: String
+    /// A string format specifier for a number, as in %d for an integer.
+    let valueTypeKey: String
+    let zero: String?
+    let one: String?
+    let two: String?
+    let few: String?
+    let many: String?
+    let other: String
+
+    private enum CodingKeys: String, CodingKey {
+      case specTypeKey = "NSStringFormatSpecTypeKey"
+      case valueTypeKey = "NSStringFormatValueTypeKey"
+
+      case zero, one, two, few, many, other
+    }
+  }
+}
+
+// - MARK: Decodable conformance
+
+extension StringsDict: Decodable {
+  // Custom CodingKeys struct to be able to parse dictionaries with dynamic keys.
+  // This is necessary, because each variable used in the formatKey is used as the key
+  // to another dictionary containing the rules for that variable.
+  private struct CodingKeys: CodingKey {
+    var intValue: Int?
+    var stringValue: String
+
+    init?(intValue: Int) {
+      self.intValue = intValue
+      self.stringValue = "\(intValue)"
+    }
+    init?(stringValue: String) {
+      self.stringValue = stringValue
+    }
+    init(key: String) {
+      self.stringValue = key
+    }
+
+    // fixed keys
+    static let formatKey = CodingKeys(key: "NSStringLocalizedFormatKey")
+    static let variableWidthRules = CodingKeys(key: "NSStringVariableWidthRuleType")
+  }
```

Comment:
Clever approach, using `struct` for `CodingKeys` for flexible keys but having `static let` for fixed values 👌